### DISCUSSION TITLE
exception WebsocketDisconnected without code

### DISCUSSION THREAD

**Discussion by vessaldaneshvar** [encode/starlette#2434] [Potential Issue]
Title:
WebSocketDisconnect KeyError on missing 'code' field in disconnect message

Description:
Issue:
When the network connection is accidentally terminated without providing a 'code' field in the disconnect message, the WebSocketDisconnect exception in the starlette library raises a KeyError, resulting in an unhandled exception.

File/Location:
[Include the file path and line number where the issue is occurring. In this case, it's starlette/websockets.py, line 109]

Exception Stack Trace:

site-packages/starlette/websockets.py", line 105, in _raise_on_disconnect
    raise WebSocketDisconnect(message["code"], message.get("reason"))
KeyError: 'code'

Expected Behavior:
The WebSocketDisconnect exception should gracefully handle cases where the 'code' field is not present in the disconnect message, avoiding the occurrence of unhandled exceptions.

Steps to Reproduce:
Describe the steps to reproduce the issue, if applicable.

Proposed Solution:
Handle the absence of the 'code' field in the disconnect message by providing a default value or checking for its existence before accessing it. This prevents the KeyError and ensures a smoother error handling process.

Environment (if applicable):

Operating system
Python version
Other relevant details
Labels (if applicable):

bug
error-handling
Priority:
Medium

Additional Information:
Any additional information or context that may help in addressing the issue.

**Comment by Kludex** [comment]
I can't read this.

Please post a code snippet I can reproduce the issue.

**Comment by vessaldaneshvar** [comment]
requirements:
`pip install wsproto==1.1.0 starlette==0.35.1 uvicorn==0.17.6 fastapi==0.109.0`
run with uvicorn:
`uvicorn --proxy-headers --ws wsproto --ws-ping-interval 30 --ws-ping-timeout 30 main:app`
source "main.py"
```from starlette.routing import WebSocketRoute, Mount
from starlette.concurrency import run_until_first_complete
from fastapi import FastAPI
import time
import json
import asyncio

async def user_ws_ping_send(websocket):
    try:
        while True:
            await websocket.send_text(json.dumps({"action": "PING"}))
            await asyncio.sleep(3)
    except asyncio.CancelledError:
        print("canceled")
    except Exception as e:
        print("ping pong error")


async def loop_ret(websocket):
    try:
      async for message in websocket.iter_json():
        print(message)

    except Exception as e:
      print(f"Connection closed: {e}")

async def websocket_handler(websocket):
    start_time = time.time()
    print(websocket._receive)
    await websocket.accept()
    try:
      await run_until_first_complete(
        (loop_ret, {"websocket": websocket}),
        (user_ws_ping_send, {"websocket": websocket}),
      )
    except Exception as e:
        print("errror until ", str(e))

    print("function properly closed", time.time()-start_time)

app = FastAPI(routes=[
    Mount("/ws", routes=[
      WebSocketRoute("/user", websocket_handler),
    ]),
])

if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app, host="0.0.0.0", port=8765)
```
if network connection break without disconnect from client, below exception raised
error : code
